### PR TITLE
Feat accordion

### LINKED ISSUE #119: Social Accordions
Social accordions são os drop-downs que irão conter as contas de cada rede social.

Esses accordions são compostos por 1 único componente, mas que possui componentes internos.

Sua estrutura é:
- Componente Header de Lista
- Componente Item de Lista

Esse componente irá receber apenas um `nome` de uma rede social e uma lista de contas dessa rede social.
Essa lista deverá conter, por enquanto, conter o nome da conta e uma imagem

Ela deve expandir e contrair de forma **animada**, ou seja, deve haver a animação de abrir e fechar.

Quando clicarmos no "titulo", ela toggle entre abrir e fechar.

Deve suportar a exibição de um icone de erro quando houver erros internos no componente.

Itens do componente:
- Icone
- Nome
- Quantiade de itens colapsados
- Setinha
- Itens dentro da lista
  - Imagem
  - Nome
  - Favoritado
  - Switch de atividade

Image

### PR DESCRIPTION
**Pull request by hxsggsz**
Closes #119 

<details open> 
  <summary>
    <b>Feature</b>
  </summary>

create social accordion component
</details>

<details open> 
  <summary>
    <b>Changelog</b>
  </summary>
create SocialAccordion folder with all necessaries files, refactor toggleSocialMedia component to follow the figma design and create a new component inside toggleSocialMedia to handle the image logic.

put `assets` folder inside `src` folder because of this error
image

</details>

<details open> 
  <summary>
    <b>Visual evidences :framed_picture:</b>
  </summary>

error variant:
image

</details>

<details open> 
  <summary>
    <b>Checklist</b>
  </summary>

  - [x] Issue linked
  - [x] Build working correctly
  - [ ] Tests created
</details>

<details open> 
  <summary>
    <b>Additional info</b>
  </summary>

because of the path alias, i couldn't create the tests for the accordion  #127 

i have to refactor how the switch component width because it was looking like this

image

</details>

### CODE REVIEW

File: src/components/SocialAccordion/SocialAccordion.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { useState } from 'react';
+
+import { AnimatePresence, motion } from 'framer-motion';
+
+import ToggleSocialMedia from '../ToggleSocialMedia/ToggleSocialMedia';
+
+import scss from './SocialAccordion.module.scss';
+
+import test from '../../../assets/logo.png';
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.tsx:9]: Esse nome aqui nao tá bom

File: src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react';
+
+import scss from './UserImage.module.scss';
+
+import { IUserImage } from './UserImage.type';
+
+function UserImage(props: IUserImage) {
+  // if account have two words it will show two letters on screen
+  // in this way will show the first one
```
Comment (aalmeida00) [inline src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx:9]: Pode remover esse comentário, o código por sí só já é explicativo.

**Review by aalmeida00 — CHANGES_REQUESTED** [review]

**Comment by hxsggsz** [comment]
ready to merge

File: src/assets/logo.png
Comment (aalmeida00) [inline src/assets/logo.png:1]: Arquivo .png é bad smell, nao consegue exportar svg?

File: src/components/SocialAccordion/SocialAccordion.module.scss [hunk 3]
```diff
@@ -0,0 +1,66 @@
+.wrapper {
+  button {
+    width: 100%;
+
+    border: 0;
+
+    background: none;
+    outline-color: #7033e0;
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.module.scss:8]: theme color

File: src/components/SocialAccordion/SocialAccordion.module.scss [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+.header {
+  display: flex;
+
+  align-items: center;
+  justify-content: space-between;
+
+  padding: 0 1.6rem;
+
+  border: 1px solid #cbc4cf;
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.module.scss:25]: theme color

File: src/components/SocialAccordion/SocialAccordion.module.scss [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    color: #7033e0;
+    font-size: 1.6rem;
+  }
+}
+
+.accordionInfo {
+  gap: 0.8rem;
+
+  span {
+    color: #49454e;
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.module.scss:50]: theme color

File: src/components/SocialAccordion/SocialAccordion.module.scss [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    font-size: 1.2rem;
+    font-weight: 500;
+  }
+
+  .error {
+    color: #c00016;
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.module.scss:56]: theme color

File: src/components/Switch/Switch.module.scss [hunk 7]
```diff
@@ -1,22 +1,22 @@
...
-  height: 28px;
+  padding: 1rem 1.8rem;
 
   position: relative;
 
   background: #c3c3c3;
   box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
-  outline: none;
-  outline: none;
+  outline-color: #7033e0;
```
Comment (aalmeida00) [inline src/components/Switch/Switch.module.scss:8]: theme color

File: src/components/ToggleSocialMedia/ToggleSocialMedia.module.scss [hunk 8]
```diff
@@ -3,9 +3,26 @@
 
   align-items: center;
   justify-content: space-between;
+
+  padding: 0.8rem;
+
+  border-bottom: 1px solid #cac4d0;
```
Comment (aalmeida00) [inline src/components/ToggleSocialMedia/ToggleSocialMedia.module.scss:9]: theme color

File: src/components/ToggleSocialMedia/ToggleSocialMedia.module.scss [hunk 9]
```diff
@@ -3,9 +3,26 @@
...
-  font-weight: 600;
+.accountInfo {
+  display: flex;
+  gap: 0.8rem;
+
+  align-items: center;
+
+  .button {
+    background: none;
+    outline-color: #7033e0;
```
Comment (aalmeida00) [inline src/components/ToggleSocialMedia/ToggleSocialMedia.module.scss:20]: theme color

File: src/components/ToggleSocialMedia/components/UserImage/UserImage.module.scss [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+.userImagePlaceholder {
+  width: 40px;
+  height: 40px;
+
+  display: grid;
+
+  place-items: center;
+
+  background-color: #eaddff;
```
Comment (aalmeida00) [inline src/components/ToggleSocialMedia/components/UserImage/UserImage.module.scss:16]: theme color

**Review by aalmeida00 — CHANGES_REQUESTED** [review]
Ajustar as variaveis de acordo com a do tema que temos.
Senti falta de um teste pro SocialAccordion.

File: src/components/SocialAccordion/SocialAccordion.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        <SocialAccordion
+          isError={false}
+          accountList={mockList}
+          socialMediaName="facebook"
+        />
+      );
+      expect(screen.getByText(/facebook/i)).toBeInTheDocument();
+    });
+
+    it('renders the intern content of accordion whe is open', () => {
```
Comment (aalmeida00) [inline src/components/SocialAccordion/SocialAccordion.spec.tsx:29]:
```suggestion
    it('renders the intern content of accordion when is open', () => {
```

File: src/components/SocialAccordion/SocialAccordion.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+              <>
+                <span>{props.accountList.length}+</span>
+                <p>{isOpen ? 'open' : 'closed'}</p>
+              </>
+            )}
+          </div>
+        </header>
+      </button>
+      <AnimatePresence>
+        {isOpen && (
```
Comment (Alecell) [inline src/components/SocialAccordion/SocialAccordion.tsx:51]: Função

File: src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+      .map((letters) => letters[0].toLocaleUpperCase());
+  }, [props.accountName]);
+
+  return props.image !== '' ? (
+    <img className={scss.userImage} src={props.image} alt="profile user" />
+  ) : (
+    <div className={scss.userImagePlaceholder}>
+      <span className={scss.initialLetter}>{getInitialLetter[0]}</span>
+    </div>
+  );
```
Comment (Alecell) [inline src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx:20]: Arrumar

**Review by aalmeida00 — CHANGES_REQUESTED** [review]

File: src/components/SocialAccordion/SocialAccordion.tsx [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+            role="listitem"
+            exit="collapsed"
+            animate="expanded"
+            initial="collapsed"
+            className={scss.list}
+            id="content-accordion"
+            variants={accordionVariants}
+            aria-labelledby="btn-accordion"
+          >
+            {props.accountList.map((accounts) => (
```
Comment (Alecell) [inline src/components/SocialAccordion/SocialAccordion.tsx:62]: Função

File: src/components/SocialAccordion/SocialAccordion.tsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+          <div className={scss.accordionInfo}>
+            {props.isError ? (
+              <span className={scss.error}>error!!!!</span>
+            ) : (
+              <>
+                <span>{props.accountList.length}+</span>
+                <p>{isOpen ? 'open' : 'closed'}</p>
+              </>
+            )}
```
Comment (Alecell) [inline src/components/SocialAccordion/SocialAccordion.tsx:46]: Função

**Review by Alecell — CHANGES_REQUESTED** [review]

**Review by aalmeida00 — APPROVED** [review]

File: src/components/SocialAccordion/SocialAccordion.tsx [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+  },
+  collapsed: { height: 0, transition: { duration: 0.3 } },
+};
+
+function SocialAccordion(props: ISocialAccordion) {
+  const [isOpen, setIsOpen] = useState(true);
+
+  const handleOpenAccordion = () => setIsOpen((prev) => !prev);
+
+  const openLabel = isOpen ? 'open' : 'closed';
```
Comment (Alecell) [inline src/components/SocialAccordion/SocialAccordion.tsx:26]: Move isso pra área de declarações de variaveis, perto do useState

File: src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    <img className={scss.userImage} src={props.image} alt="profile user" />
+  );
+
+  const renderImagePlaceholder = () => (
+    <div className={scss.userImagePlaceholder}>
+      <span className={scss.initialLetter}>{getInitialLetter[0]}</span>
+    </div>
+  );
+
+  return props.image !== '' ? renderProfileImage() : renderImagePlaceholder();
```
Comment (Alecell) [inline src/components/ToggleSocialMedia/components/UserImage/UserImage.tsx:24]:
```suggestion
  return props.image ? renderProfileImage() : renderImagePlaceholder();
```

**Review by Alecell — CHANGES_REQUESTED** [review]

**Review by Alecell — APPROVED** [review]

**Pull request merged**